ci/jenkins/test-rancher.sh

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+DEFAULT_WORKDIR=$HOME
+WORKDIR=$DEFAULT_WORKDIR
+KUBECONFIG_PATH="$HOME/.kube/config"
+TESTCASE=""
+TEST_FAILURE=false
+MODE="report"
+DOCKER_REGISTRY=$(head -n1 "${WORKSPACE}/ci/docker-registry")
+GO_VERSION=$(head -n1 "${WORKSPACE}/build/images/deps/go-version")
+IMAGE_PULL_POLICY="Always"
+
+CONFORMANCE_SKIP="\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]|\[sig-cli\]|\[sig-storage\]|\[sig-auth\]|\[sig-api-machinery\]|\[sig-apps\]|\[sig-node\]"
+NETWORKPOLICY_SKIP="should allow egress access to server in CIDR block|should enforce except clause while egress access to server in CIDR block"
+
+# In Rancher clusters the control-plane node role is "controlplane,etcd".
+CONTROL_PLANE_NODE_ROLE="controlplane,etcd"
+CLUSTER_NAME=""
+
+CLEAN_STALE_IMAGES="docker system prune --force --all --filter until=48h"
+
+_usage="Usage: $0 [--kubeconfig <KubeconfigSavePath>] [--workdir <HomePath>]
+                  [--testcase <e2e|conformance|networkpolicy>]
+
+Run K8s e2e community tests (Conformance & Network Policy) or Antrea e2e tests on a remote (Jenkins) Rancher cluster.
+
+        --cluster-name           Name of the kubernetes cluster.
+        --workdir                Home path for Go, vSphere information and antrea_logs during cluster setup. Default is $WORKDIR.
+        --testcase               Conformance and Network Policy or Antrea e2e testcases on a Rancher cluster.
+        --registry               The docker registry to use instead of dockerhub."
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --cluster-name)
+    CLUSTER_NAME="$2"
+    shift 2
+    ;;
+    --workdir)
+    WORKDIR="$2"
+    shift 2
+    ;;
+    --testcase)
+    TESTCASE="$2"
+    shift 2
+    ;;
+    --registry)
+    DOCKER_REGISTRY="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+if [[ "$WORKDIR" != "$DEFAULT_WORKDIR" && "$KUBECONFIG_PATH" == "$DEFAULT_KUBECONFIG_PATH" ]]; then
+    KUBECONFIG_PATH=${WORKDIR}/.kube/config
+fi
+NO_PULL=
+if [[ "$DOCKER_REGISTRY" != "" ]]; then
```

Comment:
I got your point. I wrongly thought the registry option could also be set in sonobuoy command line. It actually doesn't support customizing registry. Yes, it should be `"$DOCKER_REGISTRY" != ""`.

I took a look at ci/run-k8s-e2e-tests.sh, what it support is customizing sonobuoy and systemd-logs image URLs. I see that ci/run-k8s-e2e-tests.sh already provides reasonable defaults.

```
SONOBUOY_IMAGE="projects.registry.vmware.com/sonobuoy/sonobuoy:v0.56.4"
SYSTEMD_LOGS_IMAGE="projects.registry.vmware.com/sonobuoy/systemd-logs:v0.4"
```

So by default, even if `IMAGE_PULL_POLICY="Always"`, it pulls from Harbor. Since the images are with a specific tag instead of "latest", the image will not change, so I think maybe we should always use `IMAGE_PULL_POLICY="IfNotPresent"` regardless of `$DOCKER_REGISTRY` value.

I notice that the systemd-logs image URL here is different to URL in ci/run-k8s-e2e-tests.sh. Maybe we don't need to pull and re-tag systemd-logs image at all in this script.